src/components/projects/view/finalizing-activity-card.svelte

```diff
@@ -0,0 +1,124 @@
+<script>
+  import { getContext } from 'svelte';
+  import { Accordion, Button, Card } from 'attractions';
+  import { snackbarContextKey } from 'attractions/snackbar';
+  import Labeled from 'ui/labeled.svelte';
+  import EmptyState from '@/components/common/empty-state.svelte';
+  import ApplicationHourTile from '@/components/projects/view/application-hour-tile.svelte';
+  import ApplicationStatuses from '@/constants/backend/application-statuses.js';
+  import s from '@/utils/plural-s.js';
+  import * as api from '@/utils/api.js';
+  import { formatTimeRange } from '@/utils/date-time-format.js';
+
+  export let account;
+  export let activity;
+  let hourChanges = new Map();
+  let approvedApplications = activity.applications.filter(
+    apl => apl.status === ApplicationStatuses.APPROVED,
+  );
+
+  function recordHourChange({ detail: { application, hours } }) {
+    if (application.actual_hours === hours) {
+      hourChanges.delete(application);
+    } else {
+      hourChanges.set(application, hours);
+    }
+    hourChanges = hourChanges;
+  }
+
+  async function saveHours() {
+    const requests = [];
+    for (let [application, hours] of hourChanges) {
```

Comment:
```suggestion
    for (const [application, hours] of hourChanges) {
```

Reply:
I've been using `let` in loops all the time, doesn't `const` have the same scope shortcomings as `var`?

Reply:
Surely not. `const` (like `let`) creates a new block scope with each iteration. Otherwise, it would have been like reassigning to the same variable, which can't be done to a constant.